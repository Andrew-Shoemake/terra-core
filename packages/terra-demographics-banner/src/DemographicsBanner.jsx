import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import DemographicsBannerDisplay from './DemographicsBannerDisplay';
import DemographicsBannerUtils from './_sharedObjects';

const propTypes = {
  /**
   * Age of the person.
   */
  age: PropTypes.string,
  /**
   * Application content to display in the banner.
   */
  applicationContent: PropTypes.node,
  /**
   * Avatar to display in the banner. `photo` prop will be ignored if avatar is used.
   */
  avatar: PropTypes.shape(DemographicsBannerUtils.avatarShape),
  /**
   * The person's date of birth.
   */
  dateOfBirth: PropTypes.string,
  /**
   * The person's deceased date. Will display the banner as deceased if this value is provided.
   */
  deceasedDate: PropTypes.string,
  /**
   * Gender of the Person. This will be displayed on the demographics banner.
   */
  gender: PropTypes.string,
  /**
   * The gender label used for accessibility. Unlike the `Gender` prop, this label will not be displayed. It will be read in screen readers if provided.
   */
  genderAria: PropTypes.string,
  /**
   * The person's gestational age.
   */
  gestationalAge: PropTypes.string,
  /**
   * Additional key value identifiers of a person's demographic information.
   */
  // eslint-disable-next-line react/forbid-prop-types
  identifiers: PropTypes.object,
  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: intlShape.isRequired,
  /**
   * @private
   * Whether or not the banner should show active styles
   */
  isActive: PropTypes.bool,
  /**
   * Whether or not to display an icon to indicate confidentiality.
   */
  isConfidential: PropTypes.bool,
  /**
   * @private
   * Whether or not the banner should have additional styles and to indicate it is selectable.
   */
  isSelectable: PropTypes.bool,
  /**
   * Full Name of the person.
   */
  personName: PropTypes.string,
  /**
   * The column layout in which to present the displays.
   */
  postMenstrualAge: PropTypes.string,
  /**
   * The person's preferred first name if they have one.
   */
  preferredFirstName: PropTypes.string,
};

const defaultProps = {
  age: undefined,
  applicationContent: null,
  avatar: undefined,
  dateOfBirth: undefined,
  deceasedDate: null,
  gender: undefined,
  gestationalAge: null,
  identifiers: {},
  isActive: false,
  isConfidential: false,
  isSelectable: false,
  personName: undefined,
  postMenstrualAge: null,
  preferredFirstName: null,
};

const DemographicsBanner = ({
  age,
  dateOfBirth,
  gender,
  genderAria,
  intl,
  personName,
  ...customProps
}) => (
  <DemographicsBannerDisplay
    {...customProps}
    age={age}
    dateOfBirth={dateOfBirth}
    gender={gender}
    genderAria={genderAria}
    personName={personName}
    dateOfBirthLabel={intl.formatMessage({ id: 'Terra.demographicsBanner.dateOfBirth' })}
    dateOfBirthFullText={intl.formatMessage({ id: 'Terra.demographicsBanner.dateOfBirthFullText' })}
    deceasedDateLabel={intl.formatMessage({ id: 'Terra.demographicsBanner.deceased' })}
    gestationalAgeLabel={intl.formatMessage({ id: 'Terra.demographicsBanner.gestationalAge' })}
    gestationalAgeFullText={intl.formatMessage({ id: 'Terra.demographicsBanner.gestationalAgeFullText' })}
    postMenstrualAgeLabel={intl.formatMessage({ id: 'Terra.demographicsBanner.postMenstrualAge' })}
    postMenstrualAgeFullText={intl.formatMessage({ id: 'Terra.demographicsBanner.postMenstrualAgeFullText' })}
  />
);

DemographicsBanner.propTypes = propTypes;
DemographicsBanner.defaultProps = defaultProps;

export default injectIntl(DemographicsBanner);
