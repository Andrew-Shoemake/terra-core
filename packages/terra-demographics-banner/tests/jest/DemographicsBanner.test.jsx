import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import DemographicsBanner from '../../src/DemographicsBanner';

it('renders a blank banner wrapper', () => {
  const banner = shallowWithIntl(<DemographicsBanner />);
  expect(banner.dive()).toMatchSnapshot();
});

it('renders the banner wrapper with all props', () => {
  const banner = shallowWithIntl((
    <DemographicsBanner
      applicationContent={<span className="risk-score">5%</span>}
      age="25 Years"
      dateOfBirth="May 9, 1993"
      gender="Male"
      genderAria="Male"
      gestationalAge="April 5, 2016"
      identifiers={[
        {
          label: 'MRN',
          value: '12343',
          identifierAria: 'MRN 1 2 3 4 5',
        },
        {
          label: 'REA',
          value: '3JSDA',
          identifierAria: 'R E A 3JSDA',
        },
      ]}
      isConfidential
      photo=""
      personName="Johnathon Doe"
      postMenstrualAge="April 7, 2016"
      preferredFirstName="John"
    />
  ));

  expect(banner.dive()).toMatchSnapshot();
});
