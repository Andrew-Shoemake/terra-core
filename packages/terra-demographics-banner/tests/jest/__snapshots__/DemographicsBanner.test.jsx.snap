// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a blank banner wrapper 1`] = `
<InjectIntl(DemographicsBannerDisplay)
  applicationContent={null}
  dateOfBirthFullText="Terra.demographicsBanner.dateOfBirthFullText"
  dateOfBirthLabel="Terra.demographicsBanner.dateOfBirth"
  deceasedDate={null}
  deceasedDateLabel="Terra.demographicsBanner.deceased"
  gestationalAge={null}
  gestationalAgeFullText="Terra.demographicsBanner.gestationalAgeFullText"
  gestationalAgeLabel="Terra.demographicsBanner.gestationalAge"
  identifiers={Object {}}
  isConfidential={false}
  isSelectable={false}
  photo={null}
  postMenstrualAge={null}
  postMenstrualAgeFullText="Terra.demographicsBanner.postMenstrualAgeFullText"
  postMenstrualAgeLabel="Terra.demographicsBanner.postMenstrualAge"
  preferredFirstName={null}
/>
`;

exports[`renders the banner wrapper with all props 1`] = `
<InjectIntl(DemographicsBannerDisplay)
  age="25 Years"
  applicationContent={
    <span
      className="risk-score"
    >
      5%
    </span>
  }
  avatar={
    Object {
      "alt": "Johnathon Doe",
      "color": "neutral",
      "hashValue": "alternative hash",
      "image": "",
      "initials": "JD",
      "isAriaHidden": true,
    }
  }
  dateOfBirth="May 9, 1993"
  dateOfBirthFullText="Terra.demographicsBanner.dateOfBirthFullText"
  dateOfBirthLabel="Terra.demographicsBanner.dateOfBirth"
  deceasedDate={null}
  deceasedDateLabel="Terra.demographicsBanner.deceased"
  gender="Male"
  gestationalAge="April 5, 2016"
  gestationalAgeFullText="Terra.demographicsBanner.gestationalAgeFullText"
  gestationalAgeLabel="Terra.demographicsBanner.gestationalAge"
  identifiers={
    Object {
      "MRN": 12343,
      "REA": "3JSDA",
    }
  }
  isConfidential={true}
  isSelectable={false}
  personName="Johnathon Doe"
  photo=""
  postMenstrualAge="April 7, 2016"
  postMenstrualAgeFullText="Terra.demographicsBanner.postMenstrualAgeFullText"
  postMenstrualAgeLabel="Terra.demographicsBanner.postMenstrualAge"
  preferredFirstName="John"
/>
`;
